src/lib.rs

```diff
@@ -210,10 +210,14 @@ pub enum Format {
     Repeat1(Box<Format>),
     /// Repeat a format an exact number of times
     RepeatCount(Expr, Box<Format>),
-    /// Restrict a format to a sub-stream of a given number of bytes
-    Slice(Expr, Box<Format>),
-    /// Matches a format at a byte offset relative to the current stream position
-    WithRelativeOffset(Expr, Box<Format>),
+    /// Matches a format without consuming the input
+    WithInput(Box<Format>),
```

Comment:
Not quite happy with the name of this format.

Reply:
Interesting, so instead of say `Intersection(A, B)` there would be `Cat(WithInput(A), B)` :thinking:

Reply:
Yeah, maybe intersection would be better… not sure!